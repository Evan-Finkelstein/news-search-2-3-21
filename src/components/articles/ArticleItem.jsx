import React from 'react';
import PropTypes from 'prop-types';


const ArticleItem = ({ title, author, description, urlToImage }) => (
    <figure >
        <img style={{
            height: '50%',
            width: '50%',

        }} src={urlToImage} />
        <figcaption>{title}</figcaption>
        <figcaption>{author}</figcaption>
        <figcaption>{description}</figcaption>
    </figure>
);

ArticleItem.propTypes = {
    title: PropTypes.string,
    author: PropTypes.string,
    description: PropTypes.string,
};

export default ArticleItem;
