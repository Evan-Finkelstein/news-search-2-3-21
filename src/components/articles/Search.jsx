import React from 'react';
import PropTypes from 'prop-types';

const ColorControls = ({ onSearch, searchState }) => (
    <>
        <form onSubmit={onSearch}>
            <label htmlFor="search"> Search Bar
                <input

                    id="search"
                    name="search"
                    type="text"
                    onChange={searchState} /></label>
            <button>Search</button>
        </form>
    </>

);
ColorControls.propTypes = {

    onSearch: PropTypes.func.isRequired,
    searchState: PropTypes.func.isRequired,

};
export default ColorControls;