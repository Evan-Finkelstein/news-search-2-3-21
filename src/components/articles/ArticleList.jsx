import React, { Component } from 'react';
import ArticleItem from './ArticleItem';

export default class ArticleList extends Component {



    render() {
        const { articles } = this.props;
        const articleElements = articles.map(article => (
            <li key={article.title} style={{ margin: '50px' }}>
                < ArticleItem {...article} />
            </li>
        ));


        return (
            <ul >
                {articleElements}
            </ul>
        );
    }
}


// const ArticleList = ({ articles }) => (
//     <>
//         { articles.map(article => (
//             <li key={article.title} style={{ margin: '50px' }}>
//                 < ArticleItem {...article} />
//             </li>
//         ))};
//     </>

// );
// ColorControls.propTypes = {

//     onSearch: PropTypes.func.isRequired,
//     searchState: PropTypes.func.isRequired,

// };
// export default ArticleList;