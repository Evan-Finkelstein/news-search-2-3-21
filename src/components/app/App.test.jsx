import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArticleList from '../articles/ArticleList';
import ArticleItem from '../articles/ArticleItem';
import Search from '../articles/Search';
describe('App component', () => {
  afterEach(() => cleanup());

  it('renders articles', () => {
    const articles = [{
      title: '',
      author: '',
      urlToImage: '',
      description: ''
    }];
    const { asFragment } = render(<ArticleList articles={articles} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders article', () => {
    const { asFragment } = render(<ArticleItem />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders search', () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });

});
