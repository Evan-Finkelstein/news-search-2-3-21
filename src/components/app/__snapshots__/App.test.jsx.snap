// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders article 1`] = `
<DocumentFragment>
  <figure>
    <img
      style="height: 50%; width: 50%;"
    />
    <figcaption />
    <figcaption />
    <figcaption />
  </figure>
</DocumentFragment>
`;

exports[`App component renders articles 1`] = `
<DocumentFragment>
  <ul>
    <li
      style="margin: 50px;"
    >
      <figure>
        <img
          src=""
          style="height: 50%; width: 50%;"
        />
        <figcaption />
        <figcaption />
        <figcaption />
      </figure>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`App component renders search 1`] = `
<DocumentFragment>
  <form>
    <label
      for="search"
    >
       Search Bar
      <input
        id="search"
        name="search"
        type="text"
      />
    </label>
    <button>
      Search
    </button>
  </form>
</DocumentFragment>
`;
